patch/files/scripts/outside.gd

```diff
@@ -1585,13 +1585,20 @@ func _on_serviceconfirm_pressed():
 			person.levelup()
 		person.away.duration = 1 + globals.originsarray.find(person.origins)
 		person.away.at = 'training'
+#Added missing warrior buff
 	elif operation.code == 'spec':
 		if person.levelupreqs.has('code') && person.levelupreqs.code == 'specialization':
 			person.levelup()
 		globals.resources.gold -= 500
-		if person.effects.has('bodyguardeffect'): person.add_effect(globals.effectdict.bodyguardeffect, true)
-		person.spec = get_node("slaveservicepanel/serviceconfirm").get_meta('spec').code
-		if person.spec == 'bodyguard': person.add_effect(globals.effectdict.bodyguardeffect)
+	if person.effects.has('bodyguardeffect'):
+		person.add_effect(globals.effectdict.bodyguardeffect, true)
+	elif person.effects.has('warrioreffect'):
+		person.add_effect(globals.effectdict.warrioreffect, true)
+	person.spec = get_node("slaveservicepanel/serviceconfirm").get_meta('spec').code
+	if person.spec == 'bodyguard':
+		person.add_effect(globals.effectdict.bodyguardeffect)
+	elif person.spec == 'warrior':
+		person.add_effect(globals.effectdict.warrioreffect)
```

Comment:
You have to keep the offset as it was. These lines have to be offsetted more.

Because offset sets execution block in Godot. What it means is that 1) those strings are executed always, not only when getting specialization, and 2) person.away triggers when `person.spec == 'warrior'`, not every time you're training specialization.

Reply:
Alright keep that in mind next time.